tools/isopack.js

```diff
@@ -1117,7 +1117,8 @@ _.extend(Isopack.prototype, {
           // If we're going to write a legacy prelink file later, track the
           // original form of the resource object (with the source in a Buffer,
           // etc) instead of the later version.  #HardcodeJs
-          if (writeLegacyBuilds && resource.type === "source" &&
+          if (writeLegacyBuilds &&
```

Comment:
So I think you need to tweak this logic somehow:

```
        if (writeLegacyBuilds) {
          if (_.any(unibuild.resources, function (resource) {
            return resource.type === "source" && resource.extension !== "js"
              && resource.extension !== "css";
          })) {
            // This package cannot be represented as an isopack-1 Isopack
            // because it uses a file implemented by registerCompiler other than
            // the very basic JS and CSS types.
            writeLegacyBuilds = false;
          }
        }
```

The idea here is that if a package only contains JS and CSS files and static assets/head/body (or the output of "legacy source handlers", which still run at package build time and only create JS/CSS/assets/head/body), then we can publish a "fat" package that has files in the pre-1.2 format.  (This code runs only for publishing packages and for processing packages downloaded into the tropohouse (since we aren't making a separate tropohouse for new vs old packages).)

But that should *NOT* be the case if the package contains a "*.js" file but that file is being processed by ecmascript (or something else).  So we should figure out how to turn off writeLegacyBuilds in that case.

(Note that any case where there is a non-trivial source file, there must be a dependency on `isobuild:compiler-plugin`, which should make old versions of the tool not choose the package in version solver anyway.  So in a sense this is just an optimization: not creating the legacy part of the package for package versions that are never going to be looked at by legacy tools anyway. But it's an important optimization, I think.)